exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java

```diff
@@ -387,16 +378,35 @@ protected void doOnMatch(RelOptRuleCall call, Filter filterRel, Project projectR
       condition = condition.accept(reverseVisitor);
       pruneCondition = pruneCondition.accept(reverseVisitor);
 
-      if (checkForSingle && isSinglePartition && !wasAllPartitionsPruned) {
+      if (descriptor.supportsMetadataCachePruning() && !wasAllPartitionsPruned) {
         // if metadata cache file could potentially be used, then assign a proper cacheFileRoot
-        String path = "";
-        for (int j = 0; j <= maxIndex; j++) {
-          path += "/" + spInfo[j];
+        int index = -1;
+        if (!matchBitSet.isEmpty()) {
+          String path = "";
+          index = matchBitSet.length() - 1;
+
+          for (int j = 0; j < matchBitSet.length(); j++) {
+            if (!matchBitSet.get(j)) {
+              // stop at the first index with no match and use the immediate
+              // previous index
+              index = j-1;
+              break;
+            }
+          }
+          for (int j=0; j <= index; j++) {
+            path += "/" + spInfo[j];
+          }
+          cacheFileRoot = descriptor.getBaseTableLocation() + path;
```

Comment:
Actually, cacheFileRoot can be null and ParquetGroupScan will default to using the selectionRoot as the location of the cache file..here's where it decides what to use: 
https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java#L221